[RFR][NOTEST] Manual tests cleanup

valaparthvi opened the pull request:
This PR brings the following changes:
1. Polarion refactoring: add assignee, casecomponent, initialEstimate and caseimportance wherever missing. Add setup, testSteps and expectedResults for manual testcases.
2. Remove test cases related to features deprecated in `appliance.version >= 5.9`.
3. Create 5 new files `test_rest_manual.py`, `cloud/test_cloud_manual.py`, `infrastructure/test_infra_manual.py`, `configuration/test_config_manual.py` and `intelligence/reports/test_reports_manual.py` and put manual tests into the respective files.

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

jawatts commented on cfme/tests/cloud/test_cloud_manual.py:
```diff
@@ -0,0 +1,131 @@
+
+from cfme import test_requirements
+
+pytestmark = [pytest.mark.ignore_stream("5.9", "5.10", "upstream")]
```
**Required** Which version is this test valid for?

mshriver replied:
+1 - This won't get collected against anything with all streams ignored. Please see comments below first, as we may be able to remove this module and tests since they're not valid or covered by automation already.

jawatts commented on cfme/tests/cloud/test_cloud_manual.py:
```diff
@@ -0,0 +1,131 @@
+
+
+@pytest.mark.manual
+@test_requirements.discovery
```
**Required** Missing tier marker for manual tests

jawatts commented on cfme/tests/configure/test_default_views_cloud.py:
```diff
@@ -33,7 +33,7 @@ def test_default_view_cloud_reset(appliance):
 
     Polarion:
-        assignee: None
+        assignee: pvala
```
**Required** Need caseimportance and fill out initialEstimate

valaparthvi commented:
@jawatts I have added `casecomponent`, `caseimportance` and `initialEstimate` wherever missing. I have also removed some manual test cases that were recently deactivated in polarion.

mshriver commented on cfme/tests/cloud/test_cloud_manual.py:
```diff
@@ -0,0 +1,132 @@
+        casecomponent: cloud
+        caseimportance: medium
+        caseposneg: negative
+        endsin: 5.8
```
Looks like this testcase can be removed altogether if its not valid for current supported versions of CFME?

mshriver commented on cfme/tests/cloud/test_cloud_manual.py:
```diff
@@ -0,0 +1,132 @@
+@pytest.mark.manual
+@pytest.mark.tier(1)
+@test_requirements.discovery
+def test_add_cloud_provider_screen():
```
It looks like a lot of this is covered by automated tests in cfme/tests/cloud/test_providers.py.

Few comments/things to do:

1: Please re-format this docblock to remove any steps that are covered by automation already, and put any remaining test steps in the Polarion docblock like:
```
Polarion:
    testSteps:
        1. test cloud provider add with invalid hostname syntax
        2. test cloud provider with invalid security protocol
```

2. This looks like it should be parametrized by provider with a common set of steps to test.

3. Consider removing the method completely by automating whatever add form fields aren't already tested.

mshriver commented on cfme/tests/configure/test_config_manual.py:
```diff
@@ -0,0 +1,34 @@
+
+from cfme import test_requirements
+
+pytestmark = [pytest.mark.ignore_stream("5.9", "5.10", "upstream")]
```
Need to modify this to collect tests against the versions that the test is relevant for.

mshriver commented on cfme/tests/configure/test_config_manual.py:
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# pylint: skip-file
```
Why add this directive here? We want these files linted.

mshriver commented on cfme/tests/configure/test_config_manual.py:
```diff
@@ -0,0 +1,34 @@
+        title: test validate landing pages for rbac
+        testSteps:
+            1.create a new role by selecting few product features.
+              2.create a group base on the above role and the create a new
```
Check formatting here, 2/3 aren't in-line

mshriver commented on cfme/tests/configure/test_config_manual.py:
```diff
@@ -0,0 +1,34 @@
+              to my settings->visuals and check the start page entries in
+              show at login drop down list
+        expectedResults:
+            1. Display landing pages for which the user has access to
```
The expected results have to match up with the steps - I believe this should look like:
```
expectedResults:
    1. 
    2. 
    3. Only the pages that the user has access to are displayed in the selection for landing page

mshriver commented on cfme/tests/test_rest_manual.py:
```diff
@@ -0,0 +1,44 @@
+
+from cfme import test_requirements
+
+pytestmark = [pytest.mark.ignore_stream("5.9", "5.10", "upstream")]
```
Need to collect against at least one version

mshriver commented on cfme/tests/test_rest_manual.py:
```diff
@@ -0,0 +1,44 @@
+
+
+@pytest.mark.manual
+@test_requirements.rest
```
tier?

lina-nikiforova commented on cfme/tests/cloud/test_providers.py:
```diff
@@ -226,6 +134,8 @@ def test_cloud_provider_add_with_bad_credentials(provider, enable_regions):
         assignee: pvala
+        casecomponent: cloud
+        caseimportance: high
         initialEstimate: None
```
`initialEstimate` is required

lina-nikiforova commented on cfme/tests/cloud/test_providers.py:
```diff
@@ -267,6 +177,8 @@ def test_cloud_provider_crud(provider, enable_regions):
         assignee: pvala
+        casecomponent: cloud
+        caseimportance: high
         initialEstimate: None
```
`initialEstimate` is required

lina-nikiforova commented on cfme/tests/cloud_infra_common/test_rest_providers.py:
```diff
@@ -56,6 +56,8 @@ def test_query_provider_attributes(provider, provider_rest, soft_assert):
         assignee: pvala
+        casecomponent: Rest
+        caseimportance: medium
         initialEstimate: None
```
`initialEstimate` is required

lina-nikiforova commented on cfme/tests/configure/test_default_filters.py:
```diff
@@ -6,15 +6,16 @@
         assignee: pvala
         casecomponent: config
+        caseimportance: high
         initialEstimate: None
```
`initialEstimate` is required

lina-nikiforova commented on cfme/tests/configure/test_tag.py:
```diff
@@ -158,8 +160,10 @@ def test_edit_tag(self, appliance, tags):
             test_flag: rest
```
what do you think about renaming this test to `test_edit_tag_from_detail` so it will have a unique name (easier to distinguish for collection of the specific test)

lina-nikiforova commented on cfme/tests/configure/test_tag.py:
```diff
@@ -225,12 +235,6 @@ def test_create_tag_with_wrong_arguments(self, appliance):
 
     @pytest.mark.tier(3)
     @pytest.mark.meta(blockers=[BZ(1451025, forced_streams=['5.7'])])
```
Please remove the blocker if it's applicable only for `5.7z`

lina-nikiforova commented on cfme/tests/test_manual.py:
```diff
@@ -3035,23 +2934,73 @@ def test_control_import_with_incorrect_schema():
-def test_cloud_volume_types():
+@test_requirements.log_depot
+@pytest.mark.tier(2)
+def test_log_azure_settings_azure():
```
Please remove this test, it's automated
https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/configure/test_logs.py#L114

lina-nikiforova commented on cfme/tests/test_rest.py:
```diff
@@ -650,6 +684,8 @@ def test_attributes_present(appliance, collection_name):
         assignee: pvala
+        casecomponent: Rest
+        caseimportance: medium
         initialEstimate: None
```
`initialEstimate` is required

lina-nikiforova commented on cfme/tests/test_rest_manual.py:
```diff
@@ -0,0 +1,50 @@
+        casecomponent: Rest
+        caseimportance: high
+        initialEstimate: 1/30h
+        startsin: 5.10
```
Should there be `uncollectif` or `ignorestream` in this case?
